app/src/androidTest/java/org/simple/clinic/patient/PatientRepositoryAndroidTest.kt

```diff
@@ -4133,4 +4134,185 @@ class PatientRepositoryAndroidTest {
+        patientUuid: UUID,
+        phoneNumber: String?,
+        assignedFacilityId: UUID?,
+        businessId: BusinessId?
+    ) {
+      val patientProfile = TestData
+          .patientProfile(
+              patientUuid = patientUuid,
+              generatePhoneNumber = false,
+              patientPhoneNumber = phoneNumber,
+              patientAssignedFacilityId = assignedFacilityId,
+              businessId = businessId
+          )
+
+      patientRepository.save(listOf(patientProfile)).blockingAwait()
+    }
+
+    // given
+    val currentFacility = TestData.facility(uuid = UUID.fromString("c0056c11-105d-4079-80fa-c745128f7fc5"), name = "CHC Bucho")
+    val facility2 = TestData.facility(uuid = UUID.fromString("532264e6-b358-445e-8b48-c6253677db24"), name = "Test")
+    database.facilityDao().save(listOf(currentFacility, facility2))
+
+
+    val patientWithCurrentFacilityAsAssignedFacility = UUID.fromString("d47ae2e7-7453-4a6f-806e-88eb130823d8")
+    createPatientWithNumericCriteriaAndAssignedFacilityID(
+        patientUuid = patientWithCurrentFacilityAsAssignedFacility,
+        phoneNumber = "1234567890",
+        assignedFacilityId = currentFacility.uuid,
+        businessId = TestData.businessId(identifier = Identifier(
+            value = "123123",
+            type = BangladeshNationalId),
+            patientUuid = patientWithCurrentFacilityAsAssignedFacility)
+    )
+
+    val patient2WithCurrentFacilityAsAssignedFacility = UUID.fromString("23761fad-432d-4ef9-a387-1a7f1a127ba5")
+    createPatientWithNumericCriteriaAndAssignedFacilityID(
+        patientUuid = patient2WithCurrentFacilityAsAssignedFacility,
+        phoneNumber = "3456789012",
+        assignedFacilityId = currentFacility.uuid,
+        businessId = TestData.businessId(identifier = Identifier(
+            value = "123123",
+            type = BangladeshNationalId),
+            patientUuid = patient2WithCurrentFacilityAsAssignedFacility)
+    )
+
+    val patientWithAnotherFacilityAsAssignedFacility = UUID.fromString("f6cfd657-bf32-40c1-9dd2-4c956fd910a8")
+    createPatientWithNumericCriteriaAndAssignedFacilityID(
+        patientUuid = patientWithAnotherFacilityAsAssignedFacility,
+        phoneNumber = "1239382472934",
+        assignedFacilityId = facility2.uuid,
+        businessId = TestData.businessId(identifier = Identifier(
+            value = "123123",
+            type = BangladeshNationalId),
+            patientUuid = patient2WithCurrentFacilityAsAssignedFacility)
+    )
+
+    //when
+    val searchResults = patientRepository
+        .search2(NumericCriteria("12"), currentFacility.uuid)
+        .map { it.uuid }
+
+    //then
+    assertThat(searchResults)
+        .containsExactly(
+            patientWithCurrentFacilityAsAssignedFacility,
+            patient2WithCurrentFacilityAsAssignedFacility,
+            patientWithAnotherFacilityAsAssignedFacility
+        ).inOrder()
```

Comment:
Please change the test data for some of the patients to verify that identifiers that do not match the criteria do not show up in the list. Here, since all the identifiers match the search term, there is no guarantee that the query is actually functioning correctly.